ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PushQueryPublisher.java

```diff
@@ -123,45 +89,26 @@ public static PushQueryPublisher createScalablePublisher(
       final Context context,
       final SlidingWindowRateLimiter scalablePushBandRateLimiter
   ) {
-    return new PushQueryPublisher(
-        ksqlEngine,
-        serviceContext,
-        exec,
-        query,
-        analysis,
-        pushRouting,
-        context,
-        scalablePushBandRateLimiter
+    final PushRoutingOptions routingOptions = new PushQueryConfigRoutingOptions(
+        ImmutableMap.of()
     );
-  }
 
-  @Override
-  public synchronized void subscribe(final Flow.Subscriber<Collection<StreamedRow>> subscriber) {
-    final PushQueryMetadata queryMetadata;
-    if (isScalablePush) {
-      final PushRoutingOptions routingOptions = new PushQueryConfigRoutingOptions(
-          ImmutableMap.of()
-      );
+    final PushQueryConfigPlannerOptions plannerOptions = new PushQueryConfigPlannerOptions(
+        query.getSessionConfig().getConfig(false),
+        query.getSessionConfig().getOverrides());
 
-      final PushQueryConfigPlannerOptions plannerOptions = new PushQueryConfigPlannerOptions(
-              query.getSessionConfig().getConfig(false),
-              query.getSessionConfig().getOverrides());
+    scalablePushBandRateLimiter.allow(KsqlQueryType.PUSH);
 
-      scalablePushBandRateLimiter.allow(KsqlQueryType.PUSH);
+    final ScalablePushQueryMetadata pushQueryMetadata = ksqlEngine
```

Comment:
My only comment is that we're actually kicking it off before the subscribe call.  I'm not sure what the implications of that are, if any.  Here is where the inspiration for this interface is defined: https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Flow.Publisher.html#subscribe-java.util.concurrent.Flow.Subscriber-

In theory, we're not meant to unleash all of the supply until subscribe is called.  After `subscribe` is called on the publisher, `onSubscribe()` is called on the subscriber and `request(1)` is made on the subscription.  Each time it requests 1 unit of data, it returns a `Collection<StreamedRow>` of all the current entries of the queue. 

 All of this is still true except subscribe no longer kicks off the process.  I'm fine with the refactoring, but I think it might be a small breach of the API.

Reply:
Huh, I thought the query wouldn't actually get kicked off until we call `start()` (which still happens in the subscribe call). Regardless of the next question I'll ask, I'd really like clarity on when the query actually starts running.

Ok, next question: I get the theory of this whole publisher/subscriber pattern, but I couldn't quite wrap my mind around whether there's any point to it here, since we call `subscribe` _immediately_ on the returned object. Am I missing something here?

Reply:
I also think that the query is not "deployed" until the `start()` call, but please correct me if I have been wrong all these days..

For the next question, I think in our case there's little difference since `subscribe` is always triggered right after construction. In general that may not be the case, and hence it may be better deferring the construction of the actual resources until the first `subscribe` call is triggered?